src/coreclr/md/runtime/metamodel.cpp

```diff
@@ -717,7 +717,17 @@ CMiniMdBase::InitColsForTable(
                                     // should we write the data into the structure
 {
     const CMiniTableDef *pTemplate;     // Template table definition.
-    CMiniColDef pCols[9];               // The col defs to init.
+    const int MaxCols = 9;
+    typedef uint32_t markword_t;
+    BYTE tempCols[sizeof(markword_t) + MaxCols * sizeof(CMiniColDef)]; // keep aligned
```

Comment:
We use a single-byte marker everywhere in this code.  I see no reason to use four bytes just at this particular place.  This array contains 3-byte structures, so you cannot align them.

Reply:
Here is similar code in another function: https://github.com/dotnet/runtime/blob/2201016c1e13bdb9abf49e2e38cadf4ee0568df2/src/coreclr/md/enc/metamodelrw.cpp#L3635-L3638

Reply:
And here: https://github.com/dotnet/runtime/blob/4710133ea0ed60c96ea77e66aee461eadc6c6f78/src/coreclr/md/runtime/metamodel.cpp#L880-L889